### ISSUE TITLE
Huge binary size in dev profile

### ISSUE THREAD

**Issue by LucasJavaudin** [pola-rs/polars#12486]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

`src/main.rs`
```rust
use polars::prelude::*;

fn main() {
    let s = Series::new("a", [1, 2, 3, 4, 5]);
    println!("{}", s);
}
```

`Cargo.toml`
```toml
[package]
name = "polars-test"
version = "0.1.0"
edition = "2021"

[dependencies]
"polars" = { version = "0.34.2", default-features = false }
```

Build dev binary with `cargo build`.

### Issue description

When compiling a rust binary with polars code, I end up with very large binary file, for the dev profile.

With the example shown above, I get a filesize of 691 MB (vs 22 MB with release profile).

When I strip the binary of the symbols ([using `strip=true`](https://doc.rust-lang.org/cargo/reference/profiles.html#strip)), the problem disappears as I get a filesize of 30 MB (vs 9.2 MB with release profile).

I have the same problem in more complex situations. For the program I am working on, the binary size increases from ~170 MB to ~1.3 GB when I add polars as a dependency.

### Expected behavior

For the example below, the binary size should be around 30 MB.

### Installed versions

<details>

</details>

**Comment by ritchie46** [comment]
> For the example below, the binary size should be around 30 MB.

Why?

**Comment by LucasJavaudin** [comment]
From what I understood, `strip` removes the debuginfo from the binary but it has nothing to do with the functions/methods/code that is not used.
`strip=true` cannot really be used in development because it results in useless backtraces.

I was wrong saying that the binary size should be around 30 MB. I did not realise that debuginfo always take so much size, [for any library](https://users.rust-lang.org/t/why-are-debug-symbols-so-huge/81117). It seems that a binary size 23 times larger with debuginfo than without is not uncommon so there is nothing wrong with polars in that regards.

Still, I find that using `polars` results in large binaries compared to other libraries I tried. If I create a similar example than the one above but with an arrow's `Int32Array` instead of a polar's `Series`, I end up with a binary 33 times smaller. I guess that the difference might be explain by the increased complexity and number of features of polars (and cheers for that!) so there is not much to do.

Lastly, I find out that using the `debug` [setting](https://doc.rust-lang.org/cargo/reference/profiles.html#debug) can decrease significantly the binary size, without resulting in useless backtraces when used only for the dependencies.
With the following `Cargo.tom`:
```toml
[profile.dev.package."*"]
debug = false
```
the debuginfo are removed for all dependencies but not for the current crate. It results in a binary size of 57 MB.
Unfortunately, this is not useful when developing polars but this can be useful for anyone developing a library which relies on polars.

Feel free to close the issue if you agree with my analysis.
